pkg/etcd/etcd.go

```diff
@@ -1067,15 +1077,33 @@ func (e *ETCD) manageLearners(ctx context.Context) {
 			logrus.Warnf("Failed to list nodes with etcd role: %v", err)
 		}
 
+		// a map to track if a node is a member of the etcd cluster or not
+		nodeIsMember := make(map[string]bool)
+		nodesMap := make(map[string]*v1.Node)
+		for _, node := range nodes {
+			nodeIsMember[node.Name] = false
+			nodesMap[node.Name] = node
+		}
+
 		for _, member := range members.Members {
```

Comment:
Do this here, not in the other blocks below.
```suggestion
		for _, member := range members.Members {
			nodeIsMember[node.Name] = true
```